common/djangoapps/entitlements/utils.py

```diff
@@ -0,0 +1,40 @@
+from course_modes.models import CourseMode
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+def is_course_run_entitlement_fullfillable(course_run_id, compare_date, entitlement):
+    """
+    Checks that the current run meets the following criteria for an entitlement
+
+    1) Is currently running or start in the future
+    2) A User can enroll in
+    3) A User can upgrade to the entitlement mode
+
+    Arguments:
+        course_run_id (String): The id of the Course run that is being checked.
+        compare_date: The date and time that we are comparing against.  Generally the current date and time.
+        entitlement: The Entitlement that we are checking against.
+
+    Returns:
+        bool: True is the Course Run is fullfillable for the CourseEntitlement.
+    """
+    course_overview = CourseOverview.get_from_id(course_run_id)
+
+    # Only courses that have not ended will be displayed
```

Comment:
This function seems like it could be used for more than just determining what to display, should we update these comments?

Reply:
Maybe "displayed" --> "eligible to fulfill an entitlement"